dataverse/client.go

```diff
@@ -4,28 +4,37 @@ import (
 	"context"
 	"fmt"
 
+	cgschema "github.com/axone-protocol/axone-contract-schema/go/cognitarium-schema/v5"
 	dvschema "github.com/axone-protocol/axone-contract-schema/go/dataverse-schema/v5"
 	"google.golang.org/grpc"
 )
 
 type Client interface {
+	// GetResourceGovAddr returns the governance address of a resource.
+	// It queries the cognitarium to get the governance address (law-stone contract address)
+	// of a resource. The resource is identified by its DID.
 	GetResourceGovAddr(context.Context, string) (string, error)
 	ExecGov(context.Context, string, string) (interface{}, error)
 }
 
 type client struct {
-	dataverseClient dvschema.QueryClient
-	cognitariumAddr string
+	dataverseClient   dvschema.QueryClient
+	cognitariumClient cgschema.QueryClient
+	cognitariumAddr   string
 }
 
-func NewDataverseClient(ctx context.Context, dataverseClient dvschema.QueryClient) (Client, error) {
+func NewDataverseClient(ctx context.Context,
+	dataverseClient dvschema.QueryClient,
+	cognitariumClient cgschema.QueryClient,
```

Comment:
If we provide the cognitarium client we don't need to fetch the contract address. I think we could remove this function as the `NewClient` below seems well implemented.

Regarding the tests we cannot test the creation of the query clients with the grpc connection, but we can still test the fetching of the cognitarium address.

Reply:
@amimart This `NewDataverseClient` is necessary for the test. Since `client` is a private struct, I cannot create it with the mocked `dataverseClient` and `cognitariumClient` to use it in the test. Therefore, I was forced to add a function to create a client, primarily for testing purposes. 
Alternatively, I could make the `client` struct public since it only contains two gRPC clients and can be created by anyone, as there is no complex logic for creating it.

Regarding the fetching of the cognitarium address, I encountered a similar issue. Since `getCognitariumAddr` is a private function, I can't directly test it and need to go through `NewClient` for testing. Given that it depends on a gRPC connection, I can test if this function is properly called and handles gRPC connection issues correctly.

Reply:
@bdeneux Oh I see, It's because the package if `dataverse_test`. By naming it `dataverse` you'll be able to access internal elements and use `&client{}` and `getCognitariumAddr()` directly :)

Reply:
@amimart Yes, I've already try but I encountered an import cycle error since test package import mocked interface and this mocked interface import the `dataverse` package :'(. 
I will try to separated file to fix this and create an `export_test.go` file to export for the `dataverse_test` package.

Reply:
@amimart I've removed `NewDataverseClient` from the main `dataverse` package but put it in `export_test.go` to export it only for test context. Same thing for `getCognitariumAddr`, exported on test context and add corresponding test.